python/test/unit/language/test_core.py

```diff
@@ -3848,6 +3848,31 @@ def _kernel(dst, src, off, N, BLOCK_SIZE: tl.constexpr, HINT: tl.constexpr):
         assert "ld.global.v4.b32" not in ptx
 
 
+@pytest.mark.interpreter
+@pytest.mark.parametrize("has_hints", [False, True])
+def test_assume(has_hints, device):
+
+    @triton.jit
+    def _kernel(out_ptr, N: tl.constexpr, BLOCK_N: tl.constexpr, HINT: tl.constexpr):
+        current_size = N - tl.program_id(0) * BLOCK_N
+        if HINT:
+            tl.assume(current_size >= BLOCK_N)
+        if current_size >= 128:
+            tl.store(out_ptr + tl.program_id(0), current_size)
+        else:
+            tl.store(out_ptr + tl.program_id(0), current_size + 101024)
+
+    output = torch.zeros(1024 // 128, device='cuda')
+    pgm = _kernel[(1024 // 128, )](output, N=1024, BLOCK_N=128, HINT=has_hints)
+    if not is_cuda():
+        return
```

Comment:
Why is this specific to cuda?

Reply:
I intend to skip this ir check for TRITON_INTERPRET=1 mode because the pgm is None in that mode. So I copy the approach taken in the `test_vectorization_hints`, it seems only necessary to verify whether this test is running in interpret mode.

Reply:
done, only skip ir in interpreter mode.

Reply:
Thanks for fixing this! This is an useful feature for AMD GPUs too! :D

Reply:
looks like the condition fails on AMD I'm guessing because of different lowering. Can you change the test to check for llvm.assum instead? This way you can simplify the rest of the test and not even have the control flow. This will make the test more unitary

Reply:
done, simplify assume test. I keep the control flow in kernel, to demonstrate the possible use case with tl.assume.